src/assets/CollectiblesController.ts

```diff
@@ -1148,6 +1153,105 @@ export class CollectiblesController extends BaseController<
 
     this.updateNestedCollectibleState(collectibles, ALL_COLLECTIBLES_STATE_KEY);
   }
+
+  /**
+   * Update colletible isTransacted status
+   *
+   * @param address - Hex address of the collectible contract.
+   * @param tokenId - Number that represents the id of the token.
+   * @param isTransacting - Collectible transacting status.
+   * @param transactionId - Hex id of the send transaction of the collectible.
+   */
+  updateCollectibleTransactionStatus(
+    address: string,
+    tokenId: string,
+    isTransacting: boolean,
+    transactionId: string,
+  ) {
+    const { allCollectibles } = this.state;
+    const { chainId, selectedAddress } = this.config;
+    const collectibles = allCollectibles[selectedAddress]?.[chainId] || [];
+    const index: number = collectibles.findIndex(
+      (collectible) =>
+        collectible.address === address && collectible.tokenId === tokenId,
+    );
+
+    if (index === -1) {
+      return;
+    }
```

Comment:
Seems like you are doing something like this in `getCollectibleTransactionStatus` so if you rename that to `findCollectibleByAddressAndTokenId` could you reuse that here?

Reply:
I was thinking about it and since we will return a collectible in findCollectibleByAddressAndTokenId, It seems that I would need to do another iteration (filter or findByIndex) to the collectibles array, which seems to be an unnecessary iteration of the array, what do you think?

Reply:
Oh, okay, you need the index, I understand. Never mind then!